sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -189,6 +191,58 @@ def progress_metrics(self):
                     else None))),
         user=self.metrics_container.to_runner_api())
 
+  def monitoring_infos(self, transform_id):
+    """Returns the list of MonitoringInfos collected by this operation."""
+    return (self.execution_time_metrics(transform_id) +
+            self.element_count_metrics(transform_id) +
+            self.user_metrics(transform_id))
```

Comment:
Perf optimization: Maybe return generators instead of creating lists here, but you can iterate on that later.

Reply:
Ack

Reply:
I have made the FnApiResults have a metrics() and monitoring_metrics() as you have suggested. Please see updated FnApiMetrics as well